### ISSUE TITLE
consider adding char shape type (i.e. emoji)

### ISSUE THREAD

**Issue by catdad** [catdad/canvas-confetti#82]
Example code:

```javascript
function text(context, x, y, scaleX, scaleY, rotation, char) {
  context.setTransform(scaleX, 0, 0, scaleY, x, y);
  context.rotate(rotation);
  context.textAlign = 'center';
  context.font = 'normal normal 16px auto';
  context.fillText(char, 0, 0);
  context.setTransform(1,0,0,1,0,0);
}
...
text(context, fetti.x, fetti.y, Math.abs(x2 - x1) * 0.2, Math.abs(y2 - y1) * 0.2, Math.PI / 10 * fetti.wobble, fetti.shape.char);
...
confetti({
  shapes: [{ type: 'char', char: '💩' }]
});
```

I am guessing we'd need an additional font family argument.

This code has very poor performance. Any acceptable solution will need to solve that.

**Comment by catdad** [comment]
Interesting... `canvas` has a [`createPattern()`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createPattern) helper, which allows creating a pattern from an image. Then, the pattern has a [`setTransform(matrix)`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasPattern/setTransform) which allows manipulating that image directly instead of the entire canvas context. This _should_ be fast?

**Comment by catdad** [comment]
This was shipped in version 1.9.0
